src/pe/tls.rs

```diff
@@ -0,0 +1,247 @@
+use crate::error;
+use alloc::vec::Vec;
+use scroll::{Pread, Pwrite, SizeWith};
+
+use crate::pe::data_directories;
+use crate::pe::options;
+use crate::pe::section_table;
+use crate::pe::utils;
+
+/// Represents the TLS directory `IMAGE_TLS_DIRECTORY64`.
+#[repr(C)]
+#[derive(Debug, PartialEq, Copy, Clone, Default, Pread, Pwrite, SizeWith)]
+pub struct ImageTlsDirectory {
+    /// The starting address of the TLS raw data.
+    // NOTE: `u32` for 32-bit binaries, `u64` for 64-bit binaries.
+    pub start_address_of_raw_data: u64,
+    /// The ending address of the TLS raw data.
+    // NOTE: `u32` for 32-bit binaries, `u64` for 64-bit binaries.
+    pub end_address_of_raw_data: u64,
+    /// The address of the TLS index.
+    // NOTE: `u32` for 32-bit binaries, `u64` for 64-bit binaries.
+    pub address_of_index: u64,
+    /// The address of the TLS callback functions.
+    ///
+    /// Terminated by a null pointer.
+    // NOTE: `u32` for 32-bit binaries, `u64` for 64-bit binaries.
+    pub address_of_callbacks: u64,
+    /// The size of the zero fill.
+    pub size_of_zero_fill: u32,
+    /// The characteristics of the TLS.
+    pub characteristics: u32,
+}
+
+/// TLS information.
+#[repr(C)]
```

Comment:
i don't think you need `repr(c)` here, but it's possible i'm missing something

Reply:
You're right. Done.